app/graphql/resolvers/add_user_to_submission_resolver.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+class AddUserToSubmissionResolver < BaseResolver
+  include Resolvers::Submissionable
+
+  def run
+    check_submission_presence!
+
+    # check if user_email in incoming args matches to the one on the submission
+    unless matching_email(submission, @arguments&.[](:user_email))
+      raise(GraphQL::ExecutionError, 'Submission not found for this user')
+    end
+
+    #  make sure submission has no user
+    if submission.user_id
+      raise(GraphQL::ExecutionError, 'Submission already has a user')
+    end
+
+    SubmissionService.add_user_to_submission(
+      submission,
+      @context[:current_user],
```

Comment:
**suggestion:** This assumes that the user requesting this mutation is the owner of the email address that was provided as an argument. While that will be the case for the web and mobile submission flows, I think that we can remove this assumption by providing a `userID` field in the input type. What do you think?

Reply:
I see what you're saying. A malicious actor could provide a correct `userEmail` that passes validation, while also providing an incorrect `userID` to associate that artwork with a different account.

Since your code works for the currently supported submission flows, let's ignore my suggestion.